substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfInfoSectionImpl.java

```diff
@@ -196,7 +1144,7 @@ private int writeCUHeader(byte[] buffer, int p) {
             /* CU length. */
             pos += putInt(0, scratch, 0);
             /* DWARF version. */
-            pos += putShort(DW_VERSION_2, scratch, 0);
+            pos += putShort(DW_VERSION_4, scratch, 0);
```

Comment:
Shouldn't we rename this to `DW_VERSION` instead? Is there any chance to use different dwarf versions in different places?

Reply:
I am already doing that. The aranges and line sections are still written as DW_VERSION_2.  n.b. it does not cause a problem for gdb.

Still, you are right that it would be better to write all sections using a single preferred dwarf version. However, before changing anything we will need to carefully check the spec to see if updating the version of those two sections to 4 also requires changes to their format. This is probably best left for a follow-up issue.